pallets/identity/src/lib.rs

```diff
@@ -374,7 +375,10 @@ decl_module! {
             let sender_key = AccountKey::try_from(ensure_signed(origin)?.encode())?;
             let issuer = Context::current_identity_or::<Self>(&sender_key)?;
             ensure!(<DidRecords>::exists(target), Error::<T>::DidMustAlreadyExist);
-
+            T::ProtocolFee::charge_fee(
```

Comment:
The function can call `unsafe_add_cdd_claim` down the lines and that function does additional checks and can return an error.

Reply:
Please check. I think this has been resolved.